infra/utils/validator/service-worker-runtime.js

```diff
@@ -0,0 +1,57 @@
+const assert = require('assert');
```

Comment:
I'm using this in the end-to-end tests to validate all of the service worker functionality—it can check individual method calls on `WorkboxSW` with specific parameters, etc.

It borrows a bit from @gauntface's original validation work using a `vm` instance and a fake context to run the actual code.